### PR TITLE
ED25519 Support and Example Update

### PR DESCRIPTION
**Pull request by 1swaraj**

### CODE REVIEW

File: packages/private-keys/src/ED25519Format.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  type: string;
+
+  constructor(privateKey: BN) {
+    this.privateKey = privateKey;
+    this.type = "ed25519";
+  }
+
+  validatePrivateKey(privateKey: BN): boolean {
+    try {
+      nacl.box.keyPair.fromSecretKey(privateKey.toBuffer());
```
Comment (chaitanyapotti) [inline packages/private-keys/src/ED25519Format.ts:17]: cannot use toBuffer in the browser. 
Please replace it with `Buffer.from(privateKey.toString("hex", 64), "hex")`
We polyfill buffer for the browser

File: packages/private-keys/src/index.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+export { ED25519Format } from "./ED25519Format";
```
Comment (chaitanyapotti) [inline packages/private-keys/src/index.ts:1]: just do export * from "./ED25119Format"

File: packages/private-keys/test/test.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { deepStrictEqual, fail } from "assert";
+import { BN } from "bn.js";
+
```
Comment (chaitanyapotti) [inline packages/private-keys/test/test.js:3]: Add tests for secp key too

File: examples/vue-app/src/App1.vue [hunk 4]
```diff
@@ -80,7 +80,7 @@ export default {
       torusdirectsdk: undefined,
```
Comment (chaitanyapotti) [inline examples/vue-app/src/App1.vue:80]: moved to vue3, please test thoroughly the entire example

File: packages/private-keys/package.json [hunk 5]
```diff
@@ -43,7 +43,8 @@
   "dependencies": {
     "@tkey/common-types": "^6.1.0",
     "bn.js": "^5.2.1",
-    "randombytes": "^2.1.0"
+    "randombytes": "^2.1.0",
+    "tweetnacl": "^1.0.3"
```
Comment (chaitanyapotti) [inline packages/private-keys/package.json:47]: please use @toruslabs/tweetnacl-js

File: packages/private-keys/src/index.ts [hunk 6]
```diff
@@ -1,3 +1,4 @@
+export * from "./ED25519Format";
 export { default as PrivateKeysError } from "./errors";
 export { default, PRIVATE_KEY_MODULE_NAME, default as PrivateKeyModule } from "./PrivateKeys";
-export { default as SECP256k1Format } from "./SECP256K1Format";
+export { SECP256K1Format } from "./SECP256K1Format";
```
Comment (chaitanyapotti) [inline packages/private-keys/src/index.ts:4]: export * from here too

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Review by YZhenY — APPROVED** [review]
lgtm

**Pull request merged**